### PR TITLE
Add more useful error messages when update_lockable_state returns a non-ok response

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Added some real error messages to the errors returned by `update_lockable_state` as well as adding a check for no updates as that is causing confusion.

If no updates to be made:
Screen Shot 2022-08-16 at 9 56 29 AM

Some of the other message wrap a bit. It's not the prettiest but I decided to let it go as they're unlikely to occur and it is readable.
Screen Shot 2022-08-16 at 9 55 21 AM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by megcrenshaw — APPROVED** [review]
I've never seen this way of rendering translated errors! Looks great.

My one question is if we'd rather not show an error message if there's nothing to save (we can just close the dialog), but I'll leave that up to y'all.

File: apps/test/unit/code-studio/components/progress/lessonLockDialog/LessonLockDialogTest.jsx [hunk 1]
```diff
@@ -203,7 +203,7 @@ describe('LessonLockDialog with stubbed section selector', () => {
     lessonLockDataApi.saveLockState.restore();
   });
 
-  it('handleSave shows error if failed', async () => {
+  it('handleSave shows defult error if failed with no message', async () => {
```
Comment (evilcrash) [inline apps/test/unit/code-studio/components/progress/lessonLockDialog/LessonLockDialogTest.jsx]: typo: defult -> default

**Comment by davidsbailey** [comment]
> My one question is if we'd rather not show an error message if there's nothing to save (we can just close the dialog), but I'll leave that up to y'all.

+1, this seems a bit rough, but I understand if this is simply not worth doing given the longer term fix of disabling the button.

**Review by davidsbailey — APPROVED** [review]

**Comment by bethanyaconnor** [comment]
> > My one question is if we'd rather not show an error message if there's nothing to save (we can just close the dialog), but I'll leave that up to y'all.

> +1, this seems a bit rough, but I understand if this is simply not worth doing given the longer term fix of disabling the button.

Yeah, my preferred solution would probably be to disable the button if there are no changes but I think I'd have to do some refactoring and wanted to get this done asap as it sounds like this is confusing users. I made a ticket to revisit this next sprint, when I should have more time: https://codedotorg.atlassian.net/browse/LP-2430

I also thought about returning a 200 from the api if there are no changes, which would just close the dialog. I can do that if folks prefer?

File: apps/test/unit/code-studio/components/progress/lessonLockDialog/LessonLockDialogTest.jsx (hunk 1 above)
Comment (evilcrash) [reply apps/test/unit/code-studio/components/progress/lessonLockDialog/LessonLockDialogTest.jsx]: thanks, +1.

**Review by evilcrash — APPROVED** [review]

**Review by dmcavoy — COMMENTED** [review]
Thank you Bethany for getting this done so quickly!

**Comment by davidsbailey** [comment]
@evilcrash could you please sync with @jmkulwik before commenting on or approving any more pull requests. thanks!

**Pull request merged**